Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -1924,7 +1924,10 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::to_spliced)
     auto new_length_double = static_cast<double>(length) + static_cast<double>(insert_count) - static_cast<double>(actual_delete_count);
 
     // 12. If newLen > 2^53 - 1, throw a TypeError exception.
-    if (new_length_double > MAX_ARRAY_LIKE_INDEX)
+    // FIXME: ArrayCreate throws for any length > 2^32 - 1, so there's no point in letting
+    //        values up to 2^53 - 1 through (spec issue). This also prevents a potential
+    //        overflow when casting from double to size_t, which is 32 bits on x86.
+    if (new_length_double > NumericLimits<u32>::max())
         return vm.throw_completion<TypeError>(global_object, ErrorType::ArrayMaxSize);
```

Comment:
That's a good point, generally Array prototype methods allow up to `MAX_ARRAY_LIKE_INDEX` and not just 2^32 - 1, since they are supposed to be generic, so you can use them on other objects like strings that do not have this limitation, but these `toX` class of methods create arrays, so they do have to use the smaller maximum, which is the array maximum size.

Reply:
Filed https://github.com/tc39/proposal-change-array-by-copy/issues/94 for this :)